core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -483,6 +486,7 @@ private void compactSingleFile(int scheduledFile, boolean isLogFile, CompactionE
                EntryInfo info = index.getInfo(key, segment, serializedKey);
                assert info != null : "No index info found for key: " + key;
                assert info.numRecords > 0;
+               Objects.requireNonNull(info, "No index info found for key: " + key);
```

Comment:
We have this check couple lines above.

Reply:
We do, and the one above doesn't work unless assertions are enabled. I can remove the top one and convert the numRecords one as well.

Reply:
Updated